### PR TITLE
participant kinds

### PR DESCRIPTION
**Pull request by frostbyte73**

### CODE REVIEW

File: pkg/rtc/room.go [hunk 1]
```diff
@@ -417,7 +417,7 @@ func (r *Room) Join(participant types.LocalParticipant, requestSource routing.Me
 		"numParticipants", len(r.participants),
 	)
 
-	if participant.IsRecorder() && !r.protoRoom.ActiveRecording {
+	if participant.Kind() == livekit.ParticipantInfo_EGRESS && !r.protoRoom.ActiveRecording {
```
Comment (davidzhao) [inline pkg/rtc/room.go:420]: I think we'll want a backwards compatibility mode here. so that older Egress that are not setting kind would still work

Comment (frostbyte73) [reply pkg/rtc/room.go:420]: I'll check, egress 1.8.0+ requires livekit 1.5.1+ so it might be ok

Comment (frostbyte73) [reply pkg/rtc/room.go:420]: 1.8.0 is setting Kind

**Review by davidzhao — APPROVED** [review]
lgtm!

**Pull request merged**